### PR TITLE
Feature: Add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by Nelofarzabi**
### In this PR, the following were affected;

- Created class Person with the following:

 

  - [ ] Instance vars: @id, @name, and @Age.
  - [ ]  Constructor with name, age, and parent_permission as parameter. name and parent_permission are optional and have default values of "Unknown" and true.
  - [ ]  Getters for @id, @name, and @Age.
  - [ ]  Setters for @name and @Age.
  - [ ]  Private method of_age? that returns true if @Age is greater or equal to 18 and false otherwise.
  - [ ]  Public method can_use_services? that returns true if person is of age or if they have permission from parents.
  

- Created class Student with the following:

  - [ ] Inherits from Person.
  - [ ]  Constructor extends parent's constructor by adding @classroom and a parameter for it.
  - [ ]  Method play_hooky that returns "¯(ツ)/¯".
 

- Created class Teacher with the following:

  - [ ] Inherits from Person.
  - [ ]  Constructor extends parent's constructor by adding [specialization](https://github.com/specialization) and a parameter for it.
  - [ ]  Override can_use_services? so it always returns true.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,13 +1,12 @@
-require_relative './person.rb'
+require_relative 'person'
 
 class Student < Person
-  def initialize(id, age, classroom, name = "Uknown", parent_permission = true)
-    super(id, age, name, parent_permission)
+  def initialize(classroom)
+    super(age, name, parent_permission: true)
     @classroom = classroom
   end
```
Comment (V-Blaze) [inline student.rb]: - You have done great so far :+1: But there are a few changes you need to make before moving to the next milestone. :+1: 

- [ ] When you inherit from a parent class and want to call the parent class's constructor using `super`, it's a good practice to accept the arguments in the `initialize method` of the child class and then pass them down to the parent class's constructor. This ensures that the child class properly initializes its own attributes and properly passes relevant data to the parent class. So I'll suggest you accept all the arguments in the initialize method of the `Student class` Before passing it down to the super. :+1: :+1: You should update your code to reflect this. :+1: 

```sh
  def initialize(age, classroom, name = "Unknown", parent_permission = true)  
    super(age, name, parent_permission)
    @classroom = classroom
  end
```

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(specialization)
+    super(id, age, name, parent_permission: true)
+    @specialization = specialization
+  end
```
Comment (V-Blaze) [inline teacher.rb]: - [ ] Just as I suggested above, you need to accept the arguments in the initialize method before passing it down to super. I'll suggest you update this also. :+1: :100:

File: person.rb [hunk 3]
```diff
@@ -2,7 +2,7 @@ class Person
   attr_accessor :name, :age
   attr_reader :id
 
-  def initialize(id, age, name = 'Unknown', parent_permission = true)
+  def initialize(id, age, name: 'Unknown', parent_permission: true)
     @id = id
```
Comment (V-Blaze) [inline person.rb:6]: - [OPTIONAL] Following the best practices,  I'll suggest generating the id randomly, this is the best practice as ID should be unique and not null, you can have a code like this. `@id `= Random.rand(1..1000)``

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @Nelofarzabi 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

almost-there


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


## Highlights
- Created individual files for `Person`, `Student`, and `Teacher`✔✔
- Most class methods implemented correctly  :heavy_check_mark: 
- Correct git flow :heavy_check_mark: 


 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Alusp — APPROVED** [review]
Hi @Nelofarzabi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
congrats
## Highlights

- [ ] Implement classes and objects. 👍
- [ ] Implement encapsulation and inheritance with Ruby.👍
- [ ] Use Professional Readme file 👍
- [ ] Descriptive PR Summary used 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Alup in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**